<a href="https://colab.research.google.com/github/krishalvian/Pancreatic-Mass-Segmentation/blob/main/Splitting_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Splitting Data**

Splitting data digunakan untuk memisahkan antara data *training* dan data *testing*.

# **Import Data**

Menginstal library dan modul yang akan digunakan

In [6]:
!pip install -q split-folders

Menghubungkan Google Drive dengan Google Colab

In [7]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# **Import Data from Drive**

Mengakses dataset dari Google Drive

In [8]:
dataset = "/content/drive/MyDrive/Dataset"

Melakukan pembagian data dengan ratio 80% (training) dan 20% (testing)

In [9]:
#Import package split-folder
import splitfolders

#Membagi data

splitfolders.ratio(dataset,
                    output = "/content/drive/MyDrive/Dataset/Raw/Data",
                    seed = 1337,
                    ratio = (.8, .2),
                    group_prefix = None, move = False)

Copying files: 428 files [00:15, 28.01 files/s]


Terdapat 428 citra yang akan dipisahkan menjadi data *training* dan *testing*

# **Split Data**

In [12]:
import os

base_dir = '/content/drive/MyDrive/Dataset/Raw/Data'

train_dir = os.path.join(base_dir, 'train')
os.chdir(base_dir)
os.rename("val","test")
test_dir = os.path.join(base_dir, 'test')

train_data = '/content/drive/MyDrive/Dataset/Raw/Data/train/Raw'
test_data = '/content/drive/MyDrive/Dataset/Raw/Data/test/Raw'

print('total citra train:', len(os.listdir(train_data)))
print('total citra test:', len(os.listdir(test_data)))

total citra train: 342
total citra test: 86


Dari data yang sudah di split, didapatkan citra untuk training sebanyak 342 citra dan citra untuk testing sebanyak 86 citra